### DISCUSSION TITLE
Command php artisan migrate failed on Laravel Production

### DISCUSSION THREAD

**Discussion by jasperf** [deployphp/deployer#2362] [Help needed] [answered]
The command `dep deploy production` failed on migration. This due to Laravel setting not allowing migration to just run on production. You need to confirm. How could we get Deployer to bypass this if we choose to?

Error in question:

```
The command "cd /home/forge/site.com/releases/17 && php artisan migrate" failed.  
                                                                                       
  Exit Code: 1 (General error)                                                         
                                                                                       
  Host Name: production                                                                
                                                                                       
  ================                                                                     
  **************************************                                               
  *     Application In Production!     *                                               
  **************************************                                               
                                                                                       
   Do you really wish to run this command? (yes/no) [no]:                              
   > Command Canceled!      
```

**Comment by jasperf** [comment]
Read https://github.com/deployphp/deployer/pull/541 on the same issue and yes, we do not want to use force all the time.

Oddly enough we already have

```
desc('Execute artisan migrate');
task('artisan:migrate', function () {
    run('{{bin/php}} {{release_path}}/artisan migrate --force');
})->once();
```

at `vendor/deployer/deployer/recipe/laravel.php` . So why does this nor work in my case? Well we have

```
task('db:migrate', function () {
    run('cd {{release_path}} && php artisan migrate');
});
```

so without the`-- force`. Guess we should use this setup too?

 Also see `{{bin/php}}` used here. Perhaps I need to overhaul our `deploy.php` some.

**Comment by jasperf** [comment, marked as answer]
Added `--force` now to our deployment scrip as well so production now also updates right away. Slightly riskier perhaps. But because the migration issue we had a faulty upgrade. Do not want that happening either.

**Reply by antonmedv** [reply]
Maybe be command has option for not interactive mode?

**Reply by justm** [reply]
I guess @antonmedv is right. artisan has this option:
-n, --no-interaction  Do not ask any interactive question
